src/lib/components/ChainSelect.svelte

```diff
@@ -0,0 +1,156 @@
+<script lang="ts">
+	import { onMount } from 'svelte';
+
+	import Icon from './Icon.svelte';
+
+	export let chain;
+	export let syncAccount;
+
+	export let chainId = 0;
+	export let chainName = 'Loading';
+
+	let selecting = false;
+	let ref;
+	let rect;
+	onMount(() => {
+		rect = ref.getBoundingClientRect();
+	});
+
+	console.log('chain', chain);
+
+	$: if (chain)
+		chain.eth.net.getId().then((id) => {
+			console.log('Chain id', id);
+			chainId = id;
+		});
+
+	$: chainName = SUPPORTED_CHAINS.find((c) => c.chainId === chainId)?.chainName;
+
+	export const SUPPORTED_CHAINS = [
+		{
+			chainId: 1,
```

Comment:
I'd suggest not including Ethereum Mainnet since it's unlikely to ever deploy there (creating a room would cost too much in transaction fees).

Reply:
That is an interesting take.

How much would it cost to deploy the ERC-721 for a room, plus mint 2 tokens, at, say 10 gwei gas?